sljit_src/sljitConfigInternal.h

```diff
@@ -971,27 +971,41 @@ SLJIT_API_FUNC_ATTRIBUTE sljit_sw sljit_exec_offset(void *code);
 #endif /* !SLJIT_COMPILE_ASSERT */
 
 #ifndef SLJIT_FALLTHROUGH
-
-#if __cplusplus >= 201703L
-#define SLJIT_FALLTHROUGH [[__fallthrough__]];
-#elif __STDC_VERSION__ >= 202311L \
-	|| (__STDC_VERSION__ > 201710L && defined(__has_c_attribute))
-#if __STDC_VERSION__ >= 202311L || __has_c_attribute(fallthrough) >= 201904L
-#define SLJIT_FALLTHROUGH [[__fallthrough__]];
+#if defined(__cplusplus) && __cplusplus >= 201703L && \
+	defined(__has_cpp_attribute)
+/* Standards-compatible C++ variant. */
+#if __has_cpp_attribute(fallthrough)
+#define SLJIT_FALLTHROUGH [[fallthrough]];
+#endif
+#elif !defined(__cplusplus) && \
+	defined(__STDC_VERSION__) && __STDC_VERSION__ >= 202311L && \
+	defined(__has_c_attribute)
+/* Standards-compatible C variant. */
+#if __has_c_attribute(fallthrough)
+#define SLJIT_FALLTHROUGH [[fallthrough]];
 #endif
 #elif defined(__clang__)
-#if (__clang_major__ > 3 || (__clang_major__ == 3 && __clang_minor__ >= 2)) \
-	&& __cplusplus >= 201103L
+/* Clang-specific variants. */
+#if (__clang_major__ > 3 || (__clang_major__ == 3 && __clang_minor__ >= 2)) && \
+	defined(__cplusplus) && __cplusplus >= 201103L && \
+	defined(__has_cpp_attribute)
```

Comment:
not all versions of clang have __has_cpp_attribute, but all versions that target C++11 or newer support the `[[clang::fallthrough]]` attribute which is why this last defined() was not used originally and is ill advised.